geo/src/main/java/org/odk/collect/geo/SelectionMapFragment.kt

```diff
@@ -0,0 +1,416 @@
+
+            val item = itemsByFeatureId.get(featureId)
+            if (item != null) {
+                map.zoomToPoint(MapPoint(item.latitude, item.longitude), map.zoom, true)
+                map.setMarkerIcon(featureId, item.largeIcon)
+                summarySheet.setItem(item)
+                summarySheetBehavior.setState(BottomSheetBehavior.STATE_EXPANDED)
+            }
+
+            selectionMapViewModel.setSelectedItemId(featureId)
+        }
+    }
+
+    private fun onClick() {
+        if (summarySheetBehavior.getState() == BottomSheetBehavior.STATE_EXPANDED) {
+            summarySheetBehavior.setState(BottomSheetBehavior.STATE_HIDDEN)
+        }
+    }
+
+    private fun update(items: List<MappableSelectItem>) {
+        if (map == null) {
+            return
+        }
+
+        updateFeatures(items)
+
+        if (!viewportInitialized && !points.isEmpty()) {
+            map.zoomToBoundingBox(points, 0.8, false)
+            viewportInitialized = true
+        }
+    }
+
+    private fun isSummaryForGivenSubmissionDisplayed(newSubmissionId: Int): Boolean {
+        return selectionMapViewModel.getSelectedItemId() == newSubmissionId && summarySheetBehavior.state != BottomSheetBehavior.STATE_HIDDEN
+    }
+
+    private fun removeEnlargedMarkerIfExist(newSubmissionId: Int) {
+        val selectedSubmissionId = selectionMapViewModel.getSelectedItemId()
+        if (selectedSubmissionId != -1 && selectedSubmissionId != newSubmissionId) {
+            map.setMarkerIcon(
+                selectedSubmissionId,
+                itemsByFeatureId[selectedSubmissionId]!!.smallIcon
+            )
+        }
+    }
+
+    /**
+     * Clears the existing features on the map and places features for the current form's instances.
+     */
+    private fun updateFeatures(items: List<MappableSelectItem>) {
+        points.clear()
+        map.clearFeatures()
+        itemsByFeatureId.clear()
+
+        for (item in items) {
+            val point = MapPoint(item.latitude, item.longitude)
+            val featureId = map.addMarker(point, false, MapFragment.BOTTOM)
+
+            map.setMarkerIcon(
+                featureId,
+                if (featureId == selectionMapViewModel.getSelectedItemId()) item.largeIcon else item.smallIcon
+            )
+
+            itemsByFeatureId.put(featureId, item)
+            points.add(point)
+        }
+    }
+
+    companion object {
+        const val REQUEST_SELECT_ITEM = "select_item"
+        const val RESULT_SELECTED_ITEM = "selected_item"
+        const val RESULT_CREATE_NEW_ITEM = "create_new_item"
+
+        private val MAP_CENTER_KEY = "map_center"
+        private val MAP_ZOOM_KEY = "map_zoom"
+    }
+}
+
+class SelectionMapViewModel : ViewModel() {
+
+    private var mapTitle = MutableLiveData<String>()
+    private var mappableItems = MutableLiveData<List<MappableSelectItem>>(emptyList())
+    private var itemCount = MutableLiveData(0)
+    private var selectedItemId = -1
```

Comment:
Can we make it a const? (that -1) or use null to represent this state I remember you saying recently that you prefer null over -1 like here

Reply:
Oh thanks yes! I'd meant to clean this up but it escaped me.